### PR TITLE
Material dialog split

### LINKED ISSUE #168: Support MaterialAlertDialogBuilder
This is currently not released in a stable version of Material Components.

Relevant doc: https://github.com/material-components/material-components-android/blob/master/docs/components/Dialog.md

Relevant code: https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/dialog/MaterialAlertDialogBuilder.java

### PR DESCRIPTION
**Pull request by ivoberger**
As noted in the corresponding issue LouisCAD/Splitties#168 `MaterialDialogAlertBuilder` isn't available in a stable release of material components yet. I'll update this once it's out.
I'm opening this now so you can take a look.
Closes #168

### CODE REVIEW

File: modules/materialdialog/src/androidMain/kotlin/splitties/materialdialog/MaterialDialog.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * Instantiates an [MaterialAlertDialogBuilder] for the [Context], sets the passed [title], [message] and
+ * [icon], applies the [dialogConfig] lambda to it, then creates an [AlertDialog] from
+ * the builder, and returns it, so you can call [AlertDialog.show] on the created dialog.
+ */
+inline fun Context.materialDialog(
+    title: CharSequence? = null,
+    message: CharSequence? = null,
+    icon: Drawable?,
```
Comment (LouisCAD) [inline modules/materialdialog/src/androidMain/kotlin/splitties/materialdialog/MaterialDialog.kt]: `icon` should default to `null`.

Comment (IIIuminator) [reply modules/materialdialog/src/androidMain/kotlin/splitties/materialdialog/MaterialDialog.kt]: That might need to be changed in the `alertdialog` and `alertdialog-appcompat`too as I just copied the code and changed the builder

**Comment by IIIuminator** [comment]
I'll also add extensions for the extra options like `setBackground` that aren't inherited from `AlertDialog.Builder`

**Comment by IIIuminator** [comment]
And add it to the material fun pack

**Comment by IIIuminator** [comment]
> I'll also add extensions for the extra options like `setBackground` that aren't inherited from `AlertDialog.Builder`

nevermind, forgot that thats available through kotlins property access syntax

File: modules/materialdialog/src/androidMain/kotlin/splitties/materialdialog/MaterialDialog.kt (hunk 1 above)
Comment (LouisCAD) [reply modules/materialdialog/src/androidMain/kotlin/splitties/materialdialog/MaterialDialog.kt]: These splits have overloads that don't have that parameter.

**Comment by IIIuminator** [comment]
All done, I'll update `material-components` to `beta01` once its out and promote this to a full PR afterwards.

**Comment by ivoberger** [comment]
beta-01 of material components is out now. I checked out the upstream develop branch and added the materialdialogs module and changed the build.gradle.kts in line with the other build files.
It's not building for me though. Seems to be missing implementation of some test classes

**Comment by LouisCAD** [comment]
Material Components 1.1.0 has finally been released!
I'll merge this very soon, and release Splitties 3.0.0 after the changelog and doc updates are complete.

**Comment by ivoberger** [comment]
👍 
Do you need me to do something here? Like fixing the merge conflicts or would you rather do that yourself?

**Comment by LouisCAD** [comment]
For now I don't, but I'll let you know if that changes. The merge conflict is about a change in dependencies versions management, and some parts there are still WIP.

**Comment by LouisCAD** [comment]
I'm sorry it took this long to merge this, but it's finally done! I made a few naming edits. It will be called "Alert Dialog Material" to be consistent with the AppCompat version and not overlap name of dialogs that would not be alert dialogs.

**Pull request merged**